crates/daphne/src/messages/mod.rs

```diff
@@ -365,27 +365,71 @@ impl From<BatchSelector> for PartialBatchSelector {
     }
 }
 
-impl Encode for PartialBatchSelector {
-    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecError> {
-        match self {
-            Self::TimeInterval => QUERY_TYPE_TIME_INTERVAL.encode(bytes)?,
-            Self::FixedSizeByBatchId { batch_id } => {
-                QUERY_TYPE_FIXED_SIZE.encode(bytes)?;
-                batch_id.encode(bytes)?;
+impl ParameterizedEncode<DapVersion> for PartialBatchSelector {
+    fn encode_with_param(
+        &self,
+        version: &DapVersion,
+        bytes: &mut Vec<u8>,
+    ) -> Result<(), CodecError> {
+        match version {
+            DapVersion::Draft09 => {
+                match self {
+                    Self::TimeInterval => QUERY_TYPE_TIME_INTERVAL.encode(bytes)?,
+                    Self::FixedSizeByBatchId { batch_id } => {
+                        QUERY_TYPE_FIXED_SIZE.encode(bytes)?;
+                        batch_id.encode(bytes)?;
+                    }
+                };
+                Ok(())
             }
-        };
-        Ok(())
+            DapVersion::Latest => {
```

Comment:
my only concern about using 'latest' is because until the RFC is approved, there could be several 'latest' drafts.